File: apps/web/src/features/home/forYouFilters.test.ts
import { describe, expect, it } from 'vitest';
import {
  DEFAULT_FOR_YOU_FILTER,
  FOR_YOU_FILTER_OPTIONS,
  selectForYouFilter,
} from './forYouFilters.js';

describe('FOR_YOU_FILTER_OPTIONS', () => {
  it("is All / Music / Podcasts / Audiobooks, in that order — the screenshots' own wording", () => {
    expect(FOR_YOU_FILTER_OPTIONS.map((o) => o.label)).toEqual([
      'All',
      'Music',
      'Podcasts',
      'Audiobooks',
    ]);
  });

  it('uses the value "books" for the Audiobooks chip, matching the content type it filters', () => {
    expect(FOR_YOU_FILTER_OPTIONS.find((o) => o.label === 'Audiobooks')?.value).toBe('books');
  });
});

describe('selectForYouFilter', () => {
  it('selects a new filter from the default', () => {
    expect(selectForYouFilter(DEFAULT_FOR_YOU_FILTER, 'music')).toBe('music');
  });

  it('clicking the already-active chip clears back to "all"', () => {
    expect(selectForYouFilter('music', 'music')).toBe(DEFAULT_FOR_YOU_FILTER);
  });

  it('selecting a different filter replaces the previous one outright', () => {
    expect(selectForYouFilter('music', 'podcasts')).toBe('podcasts');
  });
});

File: apps/android/app/src/test/java/net/auralis/app/features/home/ForYouFiltersTest.kt
package net.auralis.app.features.home

import org.junit.Assert.assertEquals
import org.junit.Test

/**
 * Pure behaviour tests for `ForYouFilters.kt`'s chip logic — the Kotlin mirror of
 * `apps/web/src/features/home/forYouFilters.test.ts`. No Compose, no Android framework type
 * anywhere in this file or the functions under test.
 */
class ForYouFiltersTest {
    @Test
    fun `the option list is exactly All, Music, Podcasts, Audiobooks in that order`() {
        assertEquals(
            listOf("all" to "All", "music" to "Music", "podcasts" to "Podcasts", "books" to "Audiobooks"),
            FOR_YOU_FILTER_OPTIONS.map { it.value to it.label },
        )
    }

    @Test
    fun `re-clicking the active chip clears the filter back to all`() {
        assertEquals("all", selectForYouFilter(current = "music", value = "music"))
    }

    @Test
    fun `selecting a different chip replaces the current filter`() {
        assertEquals("podcasts", selectForYouFilter(current = "music", value = "podcasts"))
    }
}
